# Testing
Molecule testing would be nice but wouldn't actually prove these changes, so I did this instead for now...

## Undercloud Prepare
According to https://github.com/infrawatch/python-observabilityclient/blob/master/README.md#development and hints on https://github.com/infrawatch/osp-observability-ansible/pull/13

```
$ pwd
/home/stack/obsclient/osp-observability-ansible

$ ls -ld /usr/share/ansible/roles/spawn_container
lrwxrwxrwx. 1 root root 69 Dec  1 18:49 /usr/share/ansible/roles/spawn_container -> /home/stack/obsclient/osp-observability-ansible/roles/spawn_container

$ git reset --hard origin/csibbitt-fix-more-lint
HEAD is now at 51ea05e Fixing lint in new tasks

(undercloud) [stack@undercloud-0 osp-observability-ansible]$ git status
On branch csibbitt-fix-more-lint
Your branch is up to date with 'origin/csibbitt-fix-more-lint'.

nothing to commit, working tree clean

$ cat ~/test_params.yaml 
ceilometer_pipeline_path: "/var/lib/config-data/puppet-generated/ceilometer/etc/ceilometer/pipeline.yaml"
prometheus_remote_write:
[...]
```

## Converge 

```
$ ansible-playbook -i ~/overcloud-deploy/overcloud/tripleo-ansible-inventory.yaml -e @~/test_params.yaml playbooks/deploy_sgcore.yml 
[...]
TASK [spawn_container : Report result of container startup] **********************************************************************************************************************************
ok: [controller-0] => {                                                              
    "result": {                                     
        "actions": [                                                                                
            "pulled image quay.io/infrawatch/sg-core:latest",                                                                                                  
            "created sg-core"                                                                                                                                                                
        ],                              
[...]
                "CreateCommand": [          
                    "podman",                                                                                                                                                                 
                    "container",                                                                                                                                                             
                    "create",          
                    "--name",                               
                    "sg-core",
                    "--user",
                    "root",
                    "--restart=always",
                    "--network",
                    "host",
                    "--volume",
                    "/var/lib/osp-observability/sg-core/sgcore_config.yml:/etc/sg-core.conf.yaml:z",
                    "quay.io/infrawatch/sg-core:latest"
                ],
[...]
PLAY RECAP ***********************************************************************************************************************************************************************************
controller-0               : ok=17   changed=6    unreachable=0    failed=0    skipped=4    rescued=0    ignored=0   
```

## Verify
```
[root@controller-0 ~]# head -3 /var/lib/config-data/puppet-generated/ceilometer/etc/ceilometer/pipeline.yaml
---

sources:
[...]

[root@controller-0 ~]# systemctl status tripleo_ceilometer_agent_notification.service
● tripleo_ceilometer_agent_notification.service - ceilometer_agent_notification container
     Loaded: loaded (/etc/systemd/system/tripleo_ceilometer_agent_notification.service; enabled; vendor preset: disabled)
     Active: active (running) since Wed 2022-12-07 17:50:41 UTC; 3min 52s ago
[...]

[root@controller-0 ~]# head -3 /var/lib/osp-observability/sg-core/sgcore_config.yml 
---
plugindir: bin/
loglevel: debug


[root@controller-0 ~]# podman ps | grep sg-core
d010886f250d  quay.io/infrawatch/sg-core:latest                                                                                                                        7 minutes ago  Exited (0) Less than a second ago

[root@controller-0 ~]# podman logs sg-core | tail -1
2022-12-07 18:00:10 [ERROR] failed parsing config file [error: unmarshalling config yaml: yaml: line 9: mapping values are not allowed in this context]
```

## ~Can I still merge, please?~ Fixed!

~AFAICT this sg-core error is likely related to the fact that this hasn't merged: https://github.com/infrawatch/sg-core/pull/103, So the config file in this code is invalid. I was mainly verifying that I didn't obviously break anything, and that the lookup of the spawn_container role was working without the full path.~
(It turns out it was just a spacing issue)

--- a/roles/osp_observability/templates/sgcore_config.yml.j2
+++ b/roles/osp_observability/templates/sgcore_config.yml.j2
@@ -2,10 +2,10 @@
 plugindir: bin/
 loglevel: debug
 transports:
-  - name: socket
-    config:
-        socketaddr: {{ hostvars[inventory_hostname].ansible_host }}:{{ ceilometer_port }}
-        type: tcp
+  - name: socket
+    config:
+      socketaddr: {{ hostvars[inventory_hostname].ansible_host }}:{{ ceilometer_port }}
+      type: tcp
     handlers:
       - name: ceilometer-metrics
 applications:

